tests/test_encoding/test_count_frequency_encoder.py

```diff
@@ -145,20 +146,52 @@ def test_error_if_input_df_contains_categories_not_present_in_fit_df(
     # check that the error message matches
     assert str(record.value) == msg
 
+    # check for no error and no warning when errors equals 'encode'
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        encoder = CountFrequencyEncoder(errors="encode")
+        encoder.fit(df_enc)
+        encoder.transform(df_enc_rare)
+
 
 def test_fit_raises_error_if_df_contains_na(df_enc_na):
     # test case 4: when dataset contains na, fit method
-    with pytest.raises(ValueError):
-        encoder = CountFrequencyEncoder()
-        encoder.fit(df_enc_na)
+    for errors in ["raise", "ignore", "encode"]:
```

Comment:
why do we need this change? The NA check comes before the unseen categories check.

Reply:
It should still fail when passing "encode" if the fit input has NAs.